pkg/hostagent/hostagent.go

```diff
@@ -153,6 +153,22 @@ func New(instName string, stdout io.Writer, signalCh chan os.Signal, opts ...Opt
 		AdditionalArgs: sshutil.SSHArgsFromOpts(sshOpts),
 	}
 
+	ignoreTCP := false
+	ignoreUDP := false
+	for _, rule := range y.PortForwards {
+		if rule.Ignore && rule.GuestPortRange[0] == 1 && rule.GuestPortRange[1] == 65535 {
+			switch rule.Proto {
+			case limayaml.TCP:
+				ignoreTCP = true
+				logrus.Info("TCP port forwarding is disabled (except for SSH)")
+			case limayaml.UDP:
+				ignoreUDP = true
+				logrus.Info("UDP port forwarding is disabled")
```

Comment:
Are these really info level messages? They are usefull only for debugging your configuration - so debug level.

What if you don't specify rule.Proto? we consider this as ignore both or not ignore any?

Reply:
> Are these really info level messages? They are usefull only for debugging your configuration - so debug level.

I don't have strong opinions on this, but in some way **all** the info messages are only useful for debugging your configuration. The only one that will be left is:

```
INFO[0069] READY. Run `lima` to open the shell.
```

I would prefer to keep this at the info level, just so that if somebody puts the block to ignore port forwarding in their `default.yaml` or `override.yaml` files, and then forgets about it, they will have a reminder in the info log of what they have done.

> What if you don't specify rule.Proto? we consider this as ignore both or not ignore any?

https://github.com/lima-vm/lima/blob/52f5ad31e4c1f2a58fc37d1191ceebfdd5150dbe/pkg/limayaml/defaults.go#L809-L811

So you would need to add another rule to ignore with this PR:

```yaml
portForwards:
- ignore: true
- ignore: true
  proto: udp
```

We probably should have support for `proto: all` and default to that.

You will notice that support for UDP and IPv6 is not fully considered in the port forwarding implementation and needs to be improved. Note that support for UDP port forwarding was only added to Lima last week in #2411.

Reply:
Another issue - this works on single rule level. If you have:

```yaml
portForwards:
- ignore: true
```

This is pretty clear, but if you have:

```yaml
portForwards:
- guestPort: 80
  hostPort: 8080
- ignore: true
```

This code will treat the second rule as disabling port forwarding, while the purpose of the rule is to disable anything else port 80. In this case the user may want to see the Not forwarding message for port 80.

We can fix this by adding check for `len(y.PortForwards) == 1` (hopefully lima append the any port forwading rule later).

This could be much simpler if we have a simple switch to disable this feature. Maybe to keep the simple structrue, add something like:

```
features:
  portForwarding: false
```

With this you can keep your complex configuration as is, and enable/disable it with one change in the features map.

Another way is to make portForwards structure more flexible:

```yaml
portForwarding:
  disabled: true
  rules:
  - guestPort: x
    hostPort: y
```

It is little bit more complex but much more flexible. And it models exactly nicely the actual thing: port forwarding rules.

Reply:
> This is pretty clear, but if you have:
> 
> ```yaml
> portForwards:
> - guestPort: 80
>   hostPort: 8080
> - ignore: true
> ```
> 
> This code will treat the second rule as disabling port forwarding, while the purpose of the rule is to disable anything else port 80. In this case the user may want to see the Not forwarding message for port 80.

And they will:

```golang
for _, rule := range y.PortForwards {
	if rule.Ignore && rule.GuestPortRange[0] == 1 && rule.GuestPortRange[1] == 65535 {
		…
	} else {
		break
	}
}
```

Note how the `else { break }` part will exit the loop as soon as it encounters a rule that doesn't ignore all ports. So `ignoreTCP` and `ignoreUDP` are only set if there are no other rules in front of them.

Reply:
> > Are these really info level messages? They are usefull only for debugging your configuration - so debug level.
> 
> I don't have strong opinions on this, but in some way **all** the info messages are only useful for debugging your configuration. The only one that will be left is:
> 
> ```
> INFO[0069] READY. Run `lima` to open the shell.
> ```
> 
> I would prefer to keep this at the info level, just so that if somebody puts the block to ignore port forwarding in their `default.yaml` or `override.yaml` files, and then forgets about it, they will have a reminder in the info log of what they have done.

This is debugging why it did not work as expected, so debug level.

What can be useful in info level, is message like:

   Port forwarding is overridden by overides.yaml

And if we think that using non-default default.yaml is useful, we can log about that also. But I'm not sure about that, it depends if users had problems with such configuration.

If you configured port forwarding, info level message about forwarded port confirms that the system is operating as expected. This is also a change affecting the host, so it is import event - info level.

The messages about "waiting for retirement", "requirement satisfied" are helping the user to get some feedback about the progress. These are also useful when another program is running limactl for reporting progress. Provisioning k8s cluster takes hare about 3 minutes and having no feedback from the tool for 3 minutes is not good user experience.

Maybe the right thing is to log everything to a file, so you always have all the info, and show a nice progress with only high level info for the user.

Reply:
> everybody else who wants to wrap their head around how this can be configured

Now, if you want to argue that the documentation should be improved beyond having a bunch of samples in `default.yaml`, then I will wholeheartedly agree.

Reply:
> Maybe the right thing is to log everything to a file, so you always have all the info, and show a nice progress with only high level info for the user.

We do log everything to files (`ha.stderr.log` and `ha.stdout.log`) in the instance directory.

Personally I wouldn't mind hiding most everything until it starts waiting for requirements. Everything before is rarely useful unless you are debugging a failure:

```
INFO[0000] QEMU binary "/usr/local/bin/qemu-system-x86_64" seems properly signed with the "com.apple.security.hypervisor" entitlement
INFO[0000] Attempting to download the image              arch=x86_64 digest="sha256:0e25ca6ee9f08ec5d4f9910054b66ae7163c6152e81a3e67689d89bd6e4dfa69" location="https://cloud-images.ubuntu.com/releases/24.04/release-20240821/ubuntu-24.04-server-cloudimg-amd64.img"
INFO[0000] Using cache "/Users/jan/Library/Caches/lima/download/by-url-sha256/b2b185213b60ce48564393cf9eeb3c2ce4e92df4d2ca2edad8657c18d0e3052d/data"
INFO[0000] Attempting to download the nerdctl archive    arch=x86_64 digest="sha256:2c841e097fcfb5a1760bd354b3778cb695b44cd01f9f271c17507dc4a0b25606" location="https://github.com/containerd/nerdctl/releases/download/v1.7.6/nerdctl-full-1.7.6-linux-amd64.tar.gz"
INFO[0000] Using cache "/Users/jan/Library/Caches/lima/download/by-url-sha256/b86908f1f5ea2af45aec405f0fd389eba1999b51e3972ca78215ace94d2da2a6/data"
WARN[0001] [hostagent] GRPC port forwarding is experimental
INFO[0001] [hostagent] hostagent socket created at /Users/jan/.lima/default/ha.sock
INFO[0002] [hostagent] Using system firmware ("/usr/local/share/qemu/edk2-x86_64-code.fd")
INFO[0002] [hostagent] Starting QEMU (hint: to watch the boot progress, see "/Users/jan/.lima/default/serial*.log")
INFO[0002] SSH Local Port: 60022
```

But I think this should be a separate discussion from dropping individual `Not forwarding …` messages from the hostagent.